src/gpt_review/context.py

```diff
@@ -25,33 +23,44 @@ class Context:
 
 def _load_context_file():
     """Import from yaml file and return the context."""
-    context_file = os.getenv("CONTEXT_FILE", "azure.yaml")
+    context_file = os.getenv("CONTEXT_FILE", C.AZURE_CONFIG_FILE)
     with open(context_file, "r", encoding="utf8") as file:
         return yaml.load(file, Loader=yaml.SafeLoader)
 
 
 def _load_azure_openai_context() -> Context:
-    """
-    Load the Azure OpenAI context.
-    If the environment variables are not set, retrieve the values from Azure Key Vault.
-    Set both the environment variables and the openai package variables.
-    - Without setting the environment variables, the integration tests fail.
-    - Without setting the openai package variables, the cli tests fail.
-    """
+    """Load the context from the environment variables or the context file.
 
+    If a config file is available its values will take precedence. Otherwise
+    it will first check for an AZURE_OPENAI_API key, next OPENAI_API_KEY, and
+    lastly the Azure Key Vault.
+
+    Returns:
+        Context: The context for the Azure OpenAI API and the models.
+    """
     azure_config = _load_context_file() if os.path.exists(os.getenv("CONTEXT_FILE", C.AZURE_CONFIG_FILE)) else {}
 
-    openai.api_type = os.environ["OPENAI_API_TYPE"] = azure_config.get("azure_api_type", C.AZURE_API_TYPE)
-    openai.api_version = os.environ["OPENAI_API_VERSION"] = azure_config.get("azure_api_version", C.AZURE_API_VERSION)
+    if azure_config.get("azure_api_type"):
+        openai.api_type = os.environ["OPENAI_API_TYPE"] = azure_config.get("azure_api_type")
+    elif "OPENAI_API_TYPE" in os.environ:
+        openai.api_type = os.environ["OPENAI_API_TYPE"]
+
+    if azure_config.get("azure_api_version"):
+        openai.api_version = os.environ["OPENAI_API_VERSION"] = azure_config.get("azure_api_version")
+    elif "OPENAI_API_VERSION" in os.environ:
+        openai.api_version = os.environ["OPENAI_API_VERSION"]
```

Comment:
```suggestion
    if azure_config.get("azure_api_type"):
        openai.api_type = os.environ["OPENAI_API_TYPE"] = azure_config.get("azure_api_type")
    elif "OPENAI_API_TYPE" in os.environ:
        openai.api_type = os.environ["OPENAI_API_TYPE"]
    elif "AZURE_OPENAI_API" in os.environ:
        openai.api_type = os.environ["OPENAI_API_TYPE"] = C.AZURE_API_TYPE

    if azure_config.get("azure_api_version"):
        openai.api_version = os.environ["OPENAI_API_VERSION"] = azure_config.get("azure_api_version")
    elif "OPENAI_API_VERSION" in os.environ:
        openai.api_version = os.environ["OPENAI_API_VERSION"]
    elif "AZURE_OPENAI_API" in os.environ:
        openai.api_type = os.environ["OPENAI_API_TYPE"] = C.AZURE_API_TYPE

```

Reply:
going to circle back around later and add a more global way to toggle if you want to use azure, but for now, we want to keep the file as optional and allow for either way.